# HG changeset patch
# User Mike Miller <mtmiller@octave.org>
# Date 1473199594 25200
#      Tue Sep 06 15:06:34 2016 -0700
# Node ID 638625a0186fdde471ecc31a755a02ba454f04ff
# Parent  bb8914d68f25497afde5587e4c3c828048faefe5
Update to latest line plot color order for Matlab compatibility (bug #48465)

* graphics.cc (default_colororder): Return new default line plot color scheme.
Colors are compatible with Matlab R2014b and later versions.
* lines.m: Update doc string to reflect new default color order.
* NEWS: Mention new default line plot color order.

diff --git a/NEWS b/NEWS
--- a/NEWS
+++ b/NEWS
@@ -11,16 +11,20 @@ Summary of important user-visible change
  ** The parser has been extended to understand binary numbers which
     begin with the prefix '0b' or '0B'.  The value returned is Octave's
     default numeric class of double, not at unsigned integer class.
     Therefore numbers greater than flintmax, i.e., 2^53, will lose some
     precision.
 
     Examples: 0b101 == 5 or 0B1100_0001 == 0xC1
 
+ ** The default set of colors used to plot lines has been updated to be
+    compatible with Matlab's new default color scheme.  The line plot
+    color scheme can be set with the axes property "ColorOrder".
+
  ** The default colormap is now set to "viridis" which is also the
     default colormap in matplotlib.  This new colormap fixes some of the
     main issues with the old default colormap "jet" such as its bad
     "luminance profile" and is also more similar to Matlab's new default
     colormap "parula".
 
  ** The graphics command "hold on" now ensures that each new plot added
     to an existing plot has a different color or linestyle according to
diff --git a/libinterp/corefcn/graphics.cc b/libinterp/corefcn/graphics.cc
--- a/libinterp/corefcn/graphics.cc
+++ b/libinterp/corefcn/graphics.cc
@@ -237,34 +237,42 @@ default_screenpixelsperinch (void)
   return (display_info::x_dpi () + display_info::y_dpi ()) / 2;
 }
 
 static Matrix
 default_colororder (void)
 {
   Matrix retval (7, 3, 0.0);
 
-  retval(0,2) = 1.0;
-
-  retval(1,1) = 0.5;
-
-  retval(2,0) = 1.0;
-
-  retval(3,1) = 0.75;
-  retval(3,2) = 0.75;
-
-  retval(4,0) = 0.75;
-  retval(4,2) = 0.75;
-
-  retval(5,0) = 0.75;
-  retval(5,1) = 0.75;
-
-  retval(6,0) = 0.25;
-  retval(6,1) = 0.25;
-  retval(6,2) = 0.25;
+  retval(0,1) = 0.447;
+  retval(0,2) = 0.741;
+
+  retval(1,0) = 0.850;
+  retval(1,1) = 0.325;
+  retval(1,2) = 0.098;
+
+  retval(2,0) = 0.929;
+  retval(2,1) = 0.694;
+  retval(2,2) = 0.125;
+
+  retval(3,0) = 0.494;
+  retval(3,1) = 0.184;
+  retval(3,2) = 0.556;
+
+  retval(4,0) = 0.466;
+  retval(4,1) = 0.674;
+  retval(4,2) = 0.188;
+
+  retval(5,0) = 0.301;
+  retval(5,1) = 0.745;
+  retval(5,2) = 0.933;
+
+  retval(6,0) = 0.635;
+  retval(6,1) = 0.078;
+  retval(6,2) = 0.184;
 
   return retval;
 }
 
 static Matrix
 default_lim (bool logscale = false)
 {
   Matrix m (1, 2);
diff --git a/scripts/image/lines.m b/scripts/image/lines.m
--- a/scripts/image/lines.m
+++ b/scripts/image/lines.m
@@ -16,17 +16,17 @@
 ## along with Octave; see the file COPYING.  If not, see
 ## <http://www.gnu.org/licenses/>.
 
 ## -*- texinfo -*-
 ## @deftypefn  {} {@var{map} =} lines ()
 ## @deftypefnx {} {@var{map} =} lines (@var{n})
 ## Create color colormap.  This colormap is composed of the list of colors
 ## in the current axes @qcode{"ColorOrder"} property.  The default is blue,
-## green, red, cyan, pink, yellow, and gray.
+## orange, yellow, purple, green, light blue, and dark red.
 ##
 ## The argument @var{n} must be a scalar.
 ## If unspecified, the length of the current colormap, or 64, is used.
 ## @seealso{colormap}
 ## @end deftypefn
 
 function map = lines (n = rows (colormap ()))
 
